src/libultra/io/visetspecial.c

```diff
@@ -1,38 +1,55 @@
 #include "global.h"
+#include "ultra64/viint.h"
 
+/**
+ * Configures VI "special features" to be applied on the next context swap.
+ *
+ * "Special features" refer to the mode bits in the Video Interface control register that enable effects such as gamma
+ * correction, gamma dither, dither filtering, anti-aliasing filtering and divot filtering. Configuring the same
+ * setting ON and OFF in the same call will result in OFF taking precedence.
+ *
+ * Any unrecognized bits will be ignored. Note that this is very intentional as in early revisions of retail N64
+ * hardware setting bit 5 in the `features` field of OSViContext may cause physical damage to the console once it is
+ * fed to VI_CONTROL_REG on next context swap.
+ *
+ * @param func OS_VI_*_ON / OS_VI_*_OFF bits to enable or disable a setting.
+ */
 void osViSetSpecialFeatures(u32 func) {
     register u32 prevInt = __osDisableInt();
 
     if (func & OS_VI_GAMMA_ON) {
-        __osViNext->features |= OS_VI_GAMMA;
+        __osViNext->features |= VI_CTRL_GAMMA_ON;
     }
     if (func & OS_VI_GAMMA_OFF) {
-        __osViNext->features &= ~OS_VI_GAMMA;
+        __osViNext->features &= ~VI_CTRL_GAMMA_ON;
     }
+
     if (func & OS_VI_GAMMA_DITHER_ON) {
-        __osViNext->features |= OS_VI_GAMMA_DITHER;
+        __osViNext->features |= VI_CTRL_GAMMA_DITHER_ON;
     }
     if (func & OS_VI_GAMMA_DITHER_OFF) {
-
-        __osViNext->features &= ~OS_VI_GAMMA_DITHER;
+        __osViNext->features &= ~VI_CTRL_GAMMA_DITHER_ON;
     }
-    if (func & OS_VI_DIVOT_ON) {
 
-        __osViNext->features |= OS_VI_DIVOT;
+    if (func & OS_VI_DIVOT_ON) {
+        __osViNext->features |= VI_CTRL_DIVOT_ON;
     }
     if (func & OS_VI_DIVOT_OFF) {
-
-        __osViNext->features &= ~OS_VI_DIVOT;
+        __osViNext->features &= ~VI_CTRL_DIVOT_ON;
     }
+
     if (func & OS_VI_DITHER_FILTER_ON) {
-        __osViNext->features |= OS_VI_DITHER_FILTER;
-        __osViNext->features &= ~(OS_VI_UNK200 | OS_VI_UNK100);
+        // If the dither filter is enabled, also force anti-alias mode 0 irrespective of the current OSViMode.
+        __osViNext->features |= VI_CTRL_DITHER_FILTER_ON;
+        __osViNext->features &= ~VI_CTRL_ANTIALIAS_MASK;
```

Comment:
But is that actually an improvement? We know that frequently they set flags one at a time, this just makes it harder to read what's going on.

Reply:
I'm not sure what you mean
like,
```c
        __osViNext->features &= ~VI_CTRL_ANTIALIAS_MASK;
        __osViNext->features |= VI_CTRL_ANTIALIAS_MODE_0;
```
doesn't match

Reply:
I obviously don't understand what you're intending with this, then.

Reply:
Well the goal of this line is to set the "antialias mode" to `VI_CTRL_ANTIALIAS_MODE_0`, so for searchability/shiftability purposes (even though this relies on N64 hardware so it's not like there's a point in practice) it would be better to reference that name

Reply:
Hmm, seems like this doesn't actually make sense anyway since it's using it like an enum from the looks of it: reading the description, 3 is not actually 1 | 2 even though it looks like it?

Reply:
Yeah VI_CTRL_ANTIALIAS_MODE_0/1/2/3 is an enum indeed

Reply:
I'm reluctant to do this since it's very possible this won't match other libultra versions. (I'm also concerned about matching ultralib, particularly debug builds)

Reply:
Ok
`// VI_CTRL_ANTIALIAS_MODE_0` ? (don't have to)